### PR TITLE
Use ptr#NAME whenever the return value is stored

### PR DESCRIPTION
**Pull request by daschuer**
This is a follow up from #2601, making use of the new function

### CODE REVIEW

File: src/track/track.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ void Track::importMetadata(
         // stamps if file tags need to be re-imported.
         if (!metadataSynchronized.isNull()) {
             modified |= compareAndSet(
-                    &m_record.refMetadataSynchronized(),
+                    m_record.ptrMetadataSynchronized(),
```
Comment (uklotzde) [inline src/track/track.cpp:149]: This is the only application that I had in mind and that actually makes sense to me.

File: src/test/trackmetadata_test.cpp [hunk 2]
```diff
@@ -65,163 +65,173 @@ TEST_F(TrackMetadataTest, parseArtistTitleFromFileName) {
...
-    oldTrackInfo.setKey("1A");
-    oldTrackInfo.setReplayGain(mixxx::ReplayGain(0.1, 1));
-    oldTrackInfo.setTitle("old title");
-    oldTrackInfo.setTrackNumber("1");
-    oldTrackInfo.setTrackTotal("10");
-    oldTrackInfo.setYear("2001-01-01");
-    mixxx::AlbumInfo& oldAlbumInfo = oldTrackMetadata.refAlbumInfo();
-    oldAlbumInfo.setArtist("old album artist");
-    oldAlbumInfo.setTitle("old album title");
+    mixxx::TrackMetadata* pOldTrackMetadata = oldTrackRecord.ptrMetadata();
```
Comment (uklotzde) [inline src/test/trackmetadata_test.cpp:68]: Why all those changes???

**Review by uklotzde — CHANGES_REQUESTED** [review]
I disagree to replace all local, mutable references with pointers. That doesn't make any sense to me.

**Comment by daschuer** [comment]
Regarding non const references, there are two mutually exclusive styles established: 

1.) Google/Qt: "References can be confusing, as they have value syntax but pointer semantics."
https://google.github.io/styleguide/cppguide.html#Reference_Arguments

2.) Cpp Core Guidelines: "This makes it clear to callers that the object is assumed to be modified" 
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f17-for-in-out-parameters-pass-by-reference-to-non-const

You find a lot of threads where developer argue for one or the other. 
Let's not do It here. There are good reasons for both. 

Since we follow the Google style, I have changed the occurrences of non-const references to keep the reliability that fore instance a
a = 5; 
has no side effect.

**Comment by uklotzde** [comment]
> Regarding non const references, there are two mutually exclusive styles established:
> 
> 1.) Google/Qt: "References can be confusing, as they have value syntax but pointer semantics."
> https://google.github.io/styleguide/cppguide.html#Reference_Arguments
> 
> 2.) Cpp Core Guidelines: "This makes it clear to callers that the object is assumed to be modified"
> https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f17-for-in-out-parameters-pass-by-reference-to-non-const
> 
> You find a lot of threads where developer argue for one or the other.
> Let's not do It here. There are good reasons for both.
> 
> Since we follow the Google style, I have changed the occurrences of non-const references to keep the reliability that fore instance a
> a = 5;
> has no side effect.

I never questioned the style to pass pointers instead of mutable references for *out* arguments. This is what the citations refer to.

But where does the Google style guide refer to mutable references that are returned as results from simple, mutable accessor functions?

**Comment by daschuer** [comment]
> References can be confusing, as they have value syntax but pointer semantics.

This applies to any non-const reference variable independent if it was initialized as Funktion parameter or from a function return value.

**Comment by uklotzde** [comment]
So if I decide to store the result of a [QMap::iterator member function](https://doc.qt.io/qt-5/qmap-iterator.html) in a local variable with a descriptive name for documentation purposes and comprehension I have to apply the `&` operator to obtain a pointer??

**Comment by daschuer** [comment]
You mean ```T &iterator::value() const;```?
I think it is a good advice to always use it like ```i.value() = "Hallo"``` and not store the reference to the value at all.  
I cannot think of problem where you need to store it, because i is internally already a pointer to a QMap node and very light weight. You can even user the operator -> to alter the value directly like: 
```i->append("Hallo");```

**Comment by uklotzde** [comment]
> You mean `T &iterator::value() const;`?
> I think it is a good advice to always use it like `i.value() = "Hallo"` and not store the reference to the value at all.
> I cannot think of problem where you need to store it, because i is internally already a pointer to a QMap node and very light weight. You can even user the operator -> to alter the value directly like:
> `i->append("Hallo");`

Please let us keep those performance topics out of the discussion. They are completely irrelevant.

I am referring to ''iter.value()' which doesn't reveal neither the type nor any meaning. Binding this to a local variable helps to improve readability. Do I really have to use the awkward `&iter.value()` and bind it to a non-nullable pointer??

**Comment by daschuer** [comment]
Yes, in that case.

But I think this discussion is not relevant, because I have just skimmed through the Mixxx code and found no occourence where the return value of T& value() and friends is stored as a T&. I don't even find a place where we use value() and friends for an assignment like this ```iter.value() = "Hallo"```

This PR is just about improve the readability in the sense of the Google style:
```
- newTrackInfo.setISRC("isrc");
+ pNewTrackInfo->setISRC("isrc");
```

**Comment by uklotzde** [comment]
I am not convinced. But in order to move one I will accept these changes. They will cause some rework and resolving merge conflicts on my side now.

**Review by uklotzde — APPROVED** [review]

**Comment by daschuer** [comment]
Oh, I am sorry. 
Thank you for approval.
Merge?

**Comment by uklotzde** [comment]
Some of the indents doesn't seem to be correct, but i will fix this when needed.

**Pull request merged**